### PR TITLE
glusterd2: Refactoring of functions for snapshot

### PR DESCRIPTION
**Pull request by rafikc30**
This patch is required for supporting snapshot integration as snapshot has a dependency on volinfo and similiar functions that handles it

Signed-off-by: Mohammed Rafi KC <rkavunga@redhat.com>

### CODE REVIEW

**Comment by atinmu** [comment]
@rafikc30  Can this PR be closed in favor of https://github.com/gluster/glusterd2/pull/533 ?

**Comment by prashanthpai** [comment]
@rafikc30 Please rebase and update this PR. PR #533 is very large change as it contains 2K LOC changes over 36 files.

File: glusterd2/brick/glusterfsd.go [hunk 1]
```diff
@@ -131,3 +134,83 @@ func NewGlusterfsd(binfo Brickinfo) (*Glusterfsd, error) {
...
+
+//StopBrick will stop glusterfsd process
+func (b Brickinfo) StopBrick() error {
+
+	brickDaemon, err := NewGlusterfsd(b)
+	if err != nil {
+		return err
+	}
+
+	err = daemon.Stop(brickDaemon, true)
```
Comment (prashanthpai) [inline glusterd2/brick/glusterfsd.go]: nit: You can do ` return daemon.Stop(brickDaemon, true)` instead. Just a minor suggestion as `golint` tool run by [goreportcard](https://goreportcard.com/report/github.com/gluster/glusterd2) can catch it.

File: glusterd2/brick/types.go [hunk 2]
```diff
@@ -30,6 +30,27 @@ type Brickinfo struct {
 	Decommissioned bool
 }
 
+//For now duplicating SizeInfo till we have a common package for both brick and volume
```
Comment (prashanthpai) [inline glusterd2/brick/types.go]: We won't have a common package. You can remove this comment.

File: glusterd2/brick/validation.go [hunk 3]
```diff
@@ -52,6 +52,7 @@ func validateIsBrickMount(brickStat *unix.Stat_t, brickPath string) error {
 		return errors.ErrBrickIsMountPoint
 	}
 
+	//GfidXattrKey is used as key to set volume-id on the root of a brick
```
Comment (prashanthpai) [inline glusterd2/brick/validation.go]: Is this comment left-out my mistake ?

File: glusterd2/commands/volumes/bricks-status.go [hunk 4]
```diff
@@ -2,28 +2,43 @@ package volumecommands
...
 )
 
 const (
 	brickStatusTxnKey string = "brickstatuses"
 )
 
-func checkBricksStatus(ctx transaction.TxnCtx) error {
+func registerBricksStatusStepFuncs() {
+	transaction.RegisterStepFunc(bricksStatus, "bricks-status.Check")
+}
```
Comment (prashanthpai) [inline glusterd2/commands/volumes/bricks-status.go]: nit: Add line break after one function definition ends and another begins.

File: e2e/volume_ops_test.go [hunk 5]
```diff
@@ -51,10 +51,10 @@ func TestVolume(t *testing.T) {
 	// Run tests that depend on this volume
 	t.Run("Start", testVolumeStart)
 	t.Run("Mount", testVolumeMount)
+	t.Run("Status", testVolumeStatus)
```
Comment (prashanthpai) [inline e2e/volume_ops_test.go]: Thank you :+1:

File: glusterd2/commands/volumes/bricks-status.go [hunk 6]
```diff
@@ -35,61 +50,18 @@ func checkBricksStatus(ctx transaction.TxnCtx) error {
 		ctx.Logger().WithError(err).Error("Failed to get volume information from store.")
 		return err
 	}
-
-	mtabEntries, err := getMounts()
+	brickStatuses, err := volume.CheckBricksStatus(vol)
 	if err != nil {
-		ctx.Logger().WithError(err).Error("Failed to read /etc/mtab file.")
+		ctx.Logger().WithError(err).Error("Failed to get brick status information from store.")
```
Comment (prashanthpai) [inline glusterd2/commands/volumes/bricks-status.go]: The brick status information is not retrieved from store. You can remove the `from store` part.

File: glusterd2/commands/volumes/bricks-status.go [hunk 7]
```diff
@@ -2,28 +2,43 @@ package volumecommands
...
 
 const (
 	brickStatusTxnKey string = "brickstatuses"
 )
 
-func checkBricksStatus(ctx transaction.TxnCtx) error {
+func registerBricksStatusStepFuncs() {
+	transaction.RegisterStepFunc(bricksStatus, "bricks-status.Check")
+}
+func createBrickStatusRspAPI(brickStatuses []brick.Brickstatus) []*api.BrickStatus {
```
Comment (prashanthpai) [inline glusterd2/commands/volumes/bricks-status.go]: The name will be slightly better if this is called `createBrickStatusResp` to be consistent with rest of the code.

File: glusterd2/commands/volumes/bricks-status.go [hunk 8]
```diff
@@ -140,7 +112,7 @@ func createBricksStatusResp(ctx transaction.TxnCtx, vol *volume.Volinfo) (*api.B
 	bmap := make(map[string]*api.BrickStatus)
 	for _, b := range vol.GetBricks() {
 		bmap[b.ID.String()] = &api.BrickStatus{
-			Info: createBrickInfo(&b),
+			Info: volume.CreateBrickInfo(&b),
```
Comment (prashanthpai) [inline glusterd2/commands/volumes/bricks-status.go]: Can `CreateBrickInfo` be made part of `brick` package instead ?

File: glusterd2/commands/volumes/utils.go [hunk 9]
```diff
@@ -6,48 +6,17 @@ import (
...
-	return &api.VolumeInfo{
-		ID:        v.ID,
-		Name:      v.Name,
-		Type:      api.VolType(v.Type),
-		Transport: v.Transport,
-		DistCount: v.DistCount,
-		State:     api.VolState(v.State),
-		Options:   v.Options,
-		Subvols:   createSubvolInfo(&v.Subvols),
+func createBrickSizeInfo(size *brick.SizeInfo) api.SizeInfo {
```
Comment (prashanthpai) [inline glusterd2/commands/volumes/utils.go]: It's better is this is part of `brick` package.

File: glusterd2/xlator/validation.go [hunk 10]
```diff
@@ -94,8 +94,6 @@ func registerAllValidations() error {
 	if err := registerValidation("bit-rot", validateBitrot); err != nil {
 		return err
 	}
-	if err := registerValidation("quota", validateQuota); err != nil {
-		return err
-	}
-	return nil
+	err := registerValidation("quota", validateQuota)
```
Comment (prashanthpai) [inline glusterd2/xlator/validation.go]: You can do `registerValidation("quota", validateQuota)`

**Pull request merged**